src/circuits/etable_compact/op_configure/op_load.rs

```diff
@@ -207,6 +215,44 @@ impl<F: FieldExt> EventTableOpcodeConfigBuilder<F> for LoadConfigBuilder {
             }),
         );
 
+        constraint_builder.push(
+            "op_load value: value = padding + value_in_heap",
+            Box::new(move |meta| {
+                let padding = is_one_byte.expr(meta) * constant_from!(0xffffff00)
+                    + is_two_bytes.expr(meta) * constant_from!(0xffff0000)
+                    + (constant_from!(1) - is_eight_bytes.expr(meta)) * (vtype.expr(meta) - constant_from!(1)) * constant_from!(0xffffffff00000000);
+                vec![res.expr(meta) - value_in_heap.expr(meta) 
+                - highest_bit.expr(meta) * is_sign.expr(meta) * padding]
+            }),
+        );
+
+        constraint_builder.push(
+            "op_load highest_bit",
+            Box::new(move |meta| {
+                let mut acc32 = constant_from!(0);
+                let mut acc64 = constant_from!(0);
+
+                for i in 0..7 {
+                    acc32 = acc32
+                        + res.u4_expr(meta, i)
+                        * constant_from!(1u64 << (i * 4));
+                }
+                for i in 0..15 {
+                    acc64 = acc64
+                        + res.u4_expr(meta, i)
+                        * constant_from!(1u64 << (i * 4));
+                }
+                for i in 0..4 {
+                    acc32 = acc32
+                        + higher_four_bits[i].expr(meta) * constant_from!(1u64<<31 - i as u64); 
+                    acc64 = acc64
+                        + higher_four_bits[i].expr(meta) * constant_from!(1u64<<63 - i as u64); 
+                }
+                vec![(constant_from!(2) - vtype.expr(meta)) * (acc32 - res.expr(meta)) + (vtype.expr(meta) - constant_from!(1)) * (acc64 - res.expr(meta)),
+                 higher_four_bits[0].expr(meta) - highest_bit.expr(meta)]
```

Comment:
we can just use `higher_four_bits[0]` directly instead of use a new cell.

Reply:
right